### PR TITLE
Heron- Class!

### PR DESCRIPTION
**Pull request by spockye**
## About The Pull Request

image

This PR adds the Heron-Class Star Dreadnaught (ADMINSPAWN ONLY)!!! It's a gigantic Nanotrasen Vessel, containing a Dropship, a Mechbay, and a full Emergency Response Team!

I created this ship as a kind of "opposing force" to the Twinkleshine, though it would probably heavily differ, in use. 
while it has less soldier capacity, less combat gear, and less medical equipment, it makes up for it with its Marauder mech, Functional"tm" Supermatter engine, and its state-of-the-art, totally radical, dropship. 

the crew consists of:
___
- 1x Captain (Centcom Officer)
- 1x First Officer (hop)
- 1x HOS
- 2x Pilots (for the dropship)
- 1x Security Officer (Internal, actual Security work)
- 6x ERT Officers 
- 1x Mech Pilot
- 1x Engine Technician (Engineers)
- 1x Chief Engineer
- 1x Roboticist
- 2x Medical workers (Paramedic and Medical doctor)
- 1x Atmos tech
- 2x Cargo Crew (quartermaster and cargo tech)
- 1x Cook
- 1x Janitor
____

- [x] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game
More big ships for admins and players to have fun with is good!

## Changelog

:cl:
add: adds the Heron Class Dreadnaught
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
Screen Shot 2023-01-06 at 7 15 17 PM
criminal behavior detected. Deploying memetic kill agents

**Comment by spockye** [comment]
> Screen Shot 2023-01-06 at 7 15 17 PM
> 
> criminal behavior detected. Deploying memetic kill agents

i affirm that im going to find out where you live, and mail you large amounts of packages containing air generating devices, and dust

**Comment by Sun-Soaked** [comment]
I could be a Dust Mite for all you know. Maybe you'd be feeding me for years. Didn't think about that did you huh???

**Comment by Erikafox** [comment]
firefox_7Kgbb4Ookf
Whoever did this is evil


Ping me on the discord for a review tomorrow.

File: _maps/configs/heron.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (Erikafox) [inline _maps/configs/heron.json]: this file needs to be renamed to fit into the schema

File: _maps/configs/heron.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			"officer": true,
+			"slots": 2
+		},
+		"Security Officer": {
+			"outfit": "/datum/outfit/job/security/nanotrasen",
+			"slots": 1
+		},
+		"ERT Officer":{
+			"outfit": "/datum/outfit/job/security/nanotrasen/ert",
+			"slots": 6
```
Comment (Erikafox) [inline _maps/configs/heron.json]: I'd divide this down into having at least an engineering and medical ert fellow too.

File: code/game/area/ship_areas.dm [hunk 3]
```diff
@@ -306,6 +306,10 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 	name = "Armory"
 	icon_state = "armory"
 
+/area/ship/security/dock
+	name = "Shuttle Dock"
+	icon_state = "security"
+
```
Comment (Erikafox) [inline code/game/area/ship_areas.dm:334]: commenting here as a reminder, no need to do anything right now

File: code/modules/jobs/job_types/captain.dm [hunk 4]
```diff
@@ -62,6 +62,13 @@
 	shoes = /obj/item/clothing/shoes/laceup
 	head = /obj/item/clothing/head/caphat/nt
 
+/datum/outfit/job/captain/nt/heron
+	name = "Captain (Nanotrasen)"
```
Comment (Erikafox) [inline code/modules/jobs/job_types/captain.dm:61]: Ping a lorehead for a better name here. The centcomm clothes tell me they're a bit above a mere Captain.

File: code/modules/jobs/job_types/cargo_technician.dm [hunk 5]
```diff
@@ -65,6 +65,14 @@
...
 	glasses = /obj/item/clothing/glasses/sunglasses/big
 
+/datum/outfit/job/cargo_tech/pilot/heron
+	name = "pilot"
+
+	uniform = /obj/item/clothing/under/rank/security/officer/military
+	suit = /obj/item/clothing/suit/jacket/leather/duster
+	glasses = /obj/item/clothing/glasses/hud/spacecop
+	accessory = /obj/item/clothing/accessory/holster
+
```
Comment (Erikafox) [inline code/modules/jobs/job_types/cargo_technician.dm]: I'm starting to think we need to just wire up a pilot job.

File: _maps/configs/heron.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		},
+
+		"Atmospheric Technician": 1,
+		"Quartermaster": 1,
+		"Cargo Technician": 1,
+		"Cook": 1,
+		"Janitor": 1
+
+	},
+	"enabled": true
```
Comment (Erikafox) [inline _maps/configs/heron.json]: set false for adminspawn.

**Review by Erikafox — CHANGES_REQUESTED** [review]
The ship doesn't always spawn. 
dreamseeker_t6Ctu3boja
needs a bypass so people don't feel obligated to pull gasses out, I'd do something like...
StrongDMM_ncccgB5ozX
StrongDMM_1oc0AgM0Ed
Take the shutters out of the walls
image
swap the upper airlock for a glass one, put an air alarm in there, like the screencap.
image
the holopad is cute, but any callers will be offset from it.
StrongDMM_cwOMY1YhMJ
toss a button on the inside of the cargo area, so you can't just trap people in there
dreamseeker_5QdB05PHD3
minor decal melding, if you can figure out a way to make it have less of that, that'd be bitchin
image
Give them some suit variety, preferably to match the variety in roles (as mentioned in file review).
dreamseeker_bbIM4nukDg
Decal merging
StrongDMM_pPcmAw8Im9
Give the hardsuit it's own name, clean the captain's bathroom.
dreamseeker_iKe6DYCQQH
You should make this a sprite instead of pixelshifting 3 together.
dreamseeker_HXgU3O80NQ
You can just pull the medals off the wall, I'd recommend using a /structure for that instead of the items.
dreamseeker_BhfeS2G2fg
James (59)
StrongDMM_bGwmbM2Q6L
I would turn this area into a proper like.. Checkpoint?
StrongDMM_2T3xlzUYH9
For a big ship that I'm sure will be seeing a decent bit of downtime, there's just not enough of a crew area. I'd push it out into the hallway and giving them a bit more of a lounge - chill space.
StrongDMM_o4WEgmCOPs
You're gonna need another vent here
StrongDMM_rgI5lzXZ3o
Add some markings for the orange pipes here.
StrongDMM_MlDYSIdeGv
Same deal with the cafe, it could stand to eat into the hallway a bit
StrongDMM_k0TGMBV1ag
This airlock needs another vent too
StrongDMM_DdCOQ0Zgha
Give the AI a blast door or shutters tbh ngl
StrongDMM_4sKaa4EDZF
Box reflector in the middle, they need it if all emitters are gonna be firing.

I'm sure there'll be more but I'm feeling map fatigue, so I'll give another go over once you do all this up

File: code/modules/jobs/job_types/captain.dm (hunk 4 above)
Comment (spockye) [reply code/modules/jobs/job_types/captain.dm:61]: changed to Fleet Captain!

File: _maps/configs/heron.json (hunk 2 above)
Comment (spockye) [reply _maps/configs/heron.json]: it now has one ERT Medical Officer
and one ERT Engineering Officer
both with their respective equipment on spawn

**Comment by Erikafox** [comment]
image
If you really wanted the medals to look fancy, you could layer a notice board under them.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by Latentish** [comment]
I need to be able to see your area layout.

**Comment by Latentish** [comment]
Does that droppod actually detach?

**Comment by spockye** [comment]
> Does that droppod actually detach?

not at the moment. waiting for subshuttles to get merged

**Comment by Erikafox** [comment]
drafted like all other subship reliant prs

**Comment by goober3** [comment]
north side cargo windoor is access locked!

**Review by Erikafox — DISMISSED** [review]
Needs an exterior subshuttle port

**Comment by spockye** [comment]
ready for re-review

**Comment by Addust** [comment]
STINKY nontitanium window, replace or i will send you to god with the hyena horde
image

**Comment by spockye** [comment]
stop failing ;(

**Comment by Addust** [comment]
voidcrew beat you to it man

**Comment by Merlin1230** [comment]
weird error I found loading up this branch for shits and giggles, this ship upon loading immediately deletes itself
image
image
no clue why, doesnt seem to happen with other ships, but thought I would bring it up

**Comment by spockye** [comment]
yea, happens on my side aswell... i'll see if i can figure out why

**Comment by spockye** [comment]
seems like it's fixed, unable to reproduce it

**Comment by spockye** [comment]
Drafted until im back

**Comment by Imaginos16** [comment]
they never came back.......................

**Comment by spockye** [comment]
Erika TRAPPED ME for over a week. I'll get on fixing this now

File: code/game/area/ship_areas.dm (hunk 3 above)
Comment (goober3) [reply code/game/area/ship_areas.dm:334]: ^^ remove this area! we now have ship/hangar fulfilling the same role.

**Review by goober3 — CHANGES_REQUESTED** [review]
image
image
intercoms need pixel shifting

image
these walls need to use the new wall sprites or be replaced.

image
unrotated shutters

image
poster light

image
unconnected wire in the emitter room.

image
none of these emitters will reflect into the SM like this! please replace the reflectors with reflector boxes, or remove the outer emitters.

image
this radiation closets will get shoved out of place quite easily!! consider securing them, somehow.

image
i'd rather these high security doors be replaced with something else, for as long as they lack anywall sprites.

image
the inner windoor on the desk here should be access restricted.

image
this looks nice but please don't pixelshift holopads!! 

image
this room needs a vent & scrubber.

this ship still needs tags & a description!
every area needs a light switch. (this applies to the shuttle, too)
please replace the exterior windows with non-spawner versions.

File: _maps/shuttles/shiptest/nanotrasen_heron.dmm
Comment (meemofcourse) [inline _maps/shuttles/shiptest/nanotrasen_heron.dmm:1]: in the subshuttle dock, you are using area/hangar, and this causes an immediate sgt

Comment (spockye) [reply _maps/shuttles/shiptest/nanotrasen_heron.dmm:1]: Ill get that fixed in a day or two

**Review by goober3 — APPROVED** [review]

**Comment by goober3** [comment]
good! i'd nitpick this more but this ship been in the oven for long enough.

**Comment by Doubleumc** [comment]
Emitter reflectors start oriented such that they will reflect back at the emitter, destroying it.

No SM monitor console.

**Pull request merged**